x-pack/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/old_cluster/40_ml_datafeed_crud.yml

```diff
@@ -121,10 +124,85 @@ setup:
   - is_false: datafeeds.0.node
 
 ---
-"Put job and datafeed with aggs in old cluster - deprecated interval with warning":
+"Put job and datafeed with aggs in old cluster - deprecated interval with warning before 8.0.0":
+  - skip:
+      version: "7.99.99 - "
+      reason:  at version 8.0.0, deprecation warnings don't occur on get_stats
+      features: warnings
+
+  - do:
+      ml.put_job:
+        job_id: old-cluster-datafeed-job-with-aggs
+        body:  >
+          {
+            "description":"Cluster upgrade",
+            "analysis_config" : {
+                "bucket_span": "60s",
+                "summary_count_field_name": "doc_count",
+                "detectors" :[{"function":"count"}]
+            },
+            "analysis_limits" : {
+                "model_memory_limit": "50mb"
+            },
+            "data_description" : {
+                "format":"xcontent",
+                "time_field":"time"
+            }
+          }
+  - match: { job_id: old-cluster-datafeed-job-with-aggs }
+
+  - do:
+      warnings:
+        - '[interval] on [date_histogram] is deprecated, use [fixed_interval] or [calendar_interval] in the future.'
+      ml.put_datafeed:
+        datafeed_id: old-cluster-datafeed-with-aggs
+        body:  >
+          {
+            "job_id":"old-cluster-datafeed-job-with-aggs",
+            "indices":["airline-data"],
+            "scroll_size": 2000,
+            "aggregations": {
+              "buckets": {
+                "date_histogram": {
+                  "field": "time",
+                  "interval": "30s",
+                  "time_zone": "UTC"
+                },
+                "aggregations": {
+                  "time": {
+                    "max": {"field": "time"}
+                  },
+                  "airline": {
+                    "terms": {
+                      "field": "airline",
+                      "size": 100
+                    },
+                    "aggregations": {
+                      "responsetime": {
+                        "avg": {
+                          "field": "responsetime"
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+
+  - do:
+      warnings:
+        - '[interval] on [date_histogram] is deprecated, use [fixed_interval] or [calendar_interval] in the future.'
+      ml.get_datafeed_stats:
+        datafeed_id: old-cluster-datafeed-with-aggs
+  - match: { datafeeds.0.state: stopped}
+  - is_false: datafeeds.0.node
+
+---
+"Put job and datafeed with aggs in old cluster - deprecated interval with warning after 8.0.0":
```

Comment:
I think we will run into a problem with this in the future, because at some point before 8.0.0-beta1 is released the master branch will start to treat `interval` as a syntax error.  This test seems to be designed for the day when there is a version beyond 8.0.0 and the old cluster is 8.0.0.  But when this combination of versions is being tested this test will be forced to use `"fixed_interval" : "30s"` instead of `"interval" : "30s"`, so will not test the rewrite functionality.  Since it's also testing other things it's worth keeping the test, but you might as well change it to say `"fixed_interval" : "30s"` now.

Reply:
> This test seems to be designed for the day when there is a version beyond 8.0.0 and the old cluster is 8.0.0.

Not exactly, the rolling upgrade tests use an "old cluster" of 8.0.0, even if the "new cluster" version is 8.0.0

Reply:
This test is effectively testing the rewrite all in the same cluster version. This is actually covered when we "GET" the old datafeed in the new cluster tests. 

I will remove this particular test all together.